pkg/chunk/chunk.go

```diff
@@ -327,3 +329,47 @@ func (c *Chunk) Samples(from, through model.Time) ([]model.SamplePair, error) {
 	interval := metric.Interval{OldestInclusive: from, NewestInclusive: through}
 	return prom_chunk.RangeValues(it, interval)
 }
+
+// Slice builds a new smaller chunk with data only from given time range (inclusive)
+func (c *Chunk) Slice(from, through model.Time) (*Chunk, error) {
+	if from > c.Through || through < c.From {
+		return nil, ErrSliceOutOfRange
+	}
```

Comment:
This check allows for out of ranges as well. For example, it would allow to pass a range where `through > c.Through`. It's not necessarily a problem (it just depends on the contract of the function). I wanna be sure you're aware of that and it's by design.

Reply:
The check is to not allow passing time range which has no overlap. With some overlap, a new chunk can still be created but for no overlap, there is no chance to build a new chunk.

Reply:
Sounds good. I just wanted to make sure it was by design.